src/config.ts

```diff
@@ -22,35 +28,38 @@ export async function getDefaultConfig (server: string, token: string, owner: st
     }
 
     return await Promise.resolve(0)
-  } else {
-    try {
-      info('Restoring default config')
-      // throws an error with message containing 'Unable to find' if artifact does not exist
-      await restoreDefaultConfig()
-      info('Defaults restored from artifact')
-
-      return await Promise.resolve(0)
-    } catch (err: unknown) {
-      if (err instanceof Error && !err.message.includes('Unable to find')) {
-        throw err
-      }
-
-      // continue with downloading defaults and upload as artifact
-      info('Defaults artifact does not exist yet')
-      info('Downloading defaults')
-      await downloadDefaultConfig(server, token, owner, repository, customDefaultsPaths)
-
-      return await Promise.resolve(0)
+  }
+
+  try {
+    info('Restoring default config')
+    // throws an error with message containing 'Unable to find' if artifact does not exist
+    await restoreDefaultConfig()
+    info('Defaults restored from artifact')
+
+    return await Promise.resolve(0)
+  } catch (err: unknown) {
+    if (err instanceof Error && !err.message.includes('Unable to find')) {
+      throw err
     }
+
+    // continue with downloading defaults and upload as artifact
+    info('Defaults artifact does not exist yet')
+    info('Downloading defaults')
+    await downloadDefaultConfig(server, apiURL, version, token, owner, repository, customDefaultsPaths)
+
+    return await Promise.resolve(0)
   }
 }
 
-export async function downloadDefaultConfig (server: string, token: string, owner: string, repository: string, customDefaultsPaths: string): Promise<UploadResponse> {
+export async function downloadDefaultConfig (
+  server: string, apiURL: string, version: string, token: string, owner: string, repository: string, customDefaultsPaths: string
+): Promise<UploadResponse> {
   const customDefaultsPathsArray = customDefaultsPaths !== '' ? customDefaultsPaths.split(',') : []
-  const enterpriseDefaultsPath = getEnterpriseDefaultsUrl(owner, repository)
+
+  const [defaultsAssetURL] = await getReleaseAssetUrl(ENTERPRISE_DEFAULTS_FILENAME, version, apiURL, token, owner, repository)
```

Comment:
Local and remote name are no longer the same! The file would be named `piper-defaults-<orchestrator>.yml`

Reply:
fixed